src/dealer/GenericExchange.ts

```diff
@@ -0,0 +1,163 @@
+import { sleep } from "../utils"
+import ccxt, { ExchangeId, Order } from "ccxt"
+
+export type SupportedExchanges = "ftx" | "okex5"
+
+export class ApiConfig {
+  constructor(
+    public exchangeId: SupportedExchanges,
+    public apiKey: string,
+    public secret: string,
+    public password: string,
+  ) {
+    this.exchangeId = exchangeId
+  }
+}
+
+export class GenericExchange {
+  exchangeId: ExchangeId
+  exchange
+  public symbol
+
+  constructor(apiConfig: ApiConfig) {
+    this.exchangeId = apiConfig.exchangeId as ExchangeId
+    const exchangeClass = ccxt[this.exchangeId]
+    this.exchange = new exchangeClass({
+      apiKey: apiConfig.apiKey,
+      secret: apiConfig.secret,
+      password: apiConfig.password,
+    })
+
+    this.SetSymbol()
+
+    // Should we check at init that we have required credentials?
+    this.exchange.checkRequiredCredentials()
+  }
+
+  public SetSymbol() {
+    // TODO: Should resolve best instrument to used given the exchange and the market conditions.
+    // const optimalSymbol = this.getOptimalSymbol;
+    // this.defaultSymbol = optimalSymbol;
+
+    const exchangeMapping = {
+      ftx: "BTC-PERP",
+      okexv5: "BTC-USD-211231",
+    }
+
+    this.symbol = exchangeMapping[this.exchangeId]
+  }
+
+  public has() {
+    const has = this.exchange.has
+    return has
+  }
+
+  public has2() {
+    const has = this.exchange.has
+    return has
+  }
+
+  public name() {
+    const name = this.exchange.name
+    return name
```

Comment:
`return this.exchange.name` ? we want to have concise code. there is no need to define a const here.

Reply:
Done and updated other similar code